### DISCUSSION TITLE
processor_regex中是否使用KeepSource会影响输出数据格式？

### DISCUSSION THREAD

**Discussion by q121102302** [alibaba/ilogtail#734] [Help] [answered]
使用正则(processor_regex) 输出数据会额外打印一个tags字段，
但是在使用KeepSource=true后 这个字段就没有了 tags中包含的host.ip和host.name还是很有用的
请问应该怎么加入？

**Comment by messixukejia** [comment]
完整的配置发下看看

**Reply by q121102302** [reply]
原来的配置是这样的
<img width="791" alt="image" src="https://user-images.githubusercontent.com/101638329/226808795-7e33cf21-743f-4b94-a72b-3875187de41b.png">

输出大概为
<img width="926" alt="image" src="https://user-images.githubusercontent.com/101638329/226808723-d88c5544-a668-46f1-a6b3-c8fe47c54fed.png">

为了保证日志完整现实，并且又需要提取到信息。所以使用了KeepSource=true的配置。其他基本没变
<img width="865" alt="image" src="https://user-images.githubusercontent.com/101638329/226809196-c8d728ea-f2d2-4457-acfd-acaf75754d20.png">
打印出的日志
<img width="1083" alt="image" src="https://user-images.githubusercontent.com/101638329/226809411-837df19c-3403-4aee-b12b-9f7c5337f9d7.png">
不再具有之前的tags字段了 即少了host.ip和host.name字段

**Comment by q121102302** [comment, marked as answer]
找到原因了 输出时 输出给kafka和打印的 数据样式 不是一致的 导致的 跟这个没关系
